server/login/views.py

```diff
@@ -0,0 +1,234 @@
+        response = Response(login_failed, status=status.HTTP_401_UNAUTHORIZED)
+
+        try:
+            user = CustomUser.objects.get(email=receiver_email)
+            if user.is_active:
+                org_id = user.organization.org_id
+                org_name = user.organization.org_name
+                data = {'user': user.user_name, 'user_id': user.id, 'role': user.role,
+                        'organization_id': org_id,
+                        'organization_name': org_name}
+
+                # Update the Password as the new PIN and send an email
+                pin = save_new_pin(receiver_email, user)
+                sender_email = os.getenv('SENDER_EMAIL', 'email@email.com')
+                sender_password = os.getenv('SENDER_PASSWORD', 'pass1234')
+                send_email(
+                    sender_email,
+                    sender_password,
+                    receiver_email,
+                    pin)
+                response = Response(data, status=status.HTTP_200_OK)
+
+        except ObjectDoesNotExist as exception:
+            logger.error(exception)
+
+        return response
+
+
+def save_new_pin(email, user):
+    first_part = ''.join(secrets.choice(string.ascii_letters + string.digits)
+                         for _i in range(3)) #NOSONAR
+    second_part = '-'
+    third_part = ''.join(secrets.choice(string.ascii_letters + string.digits)
+                         for _i in range(3)) #NOSONAR
+    request = HttpRequest()
+    request.data = {'password': first_part + second_part + third_part}
+    request.user = email
+    kwargs = {'partial': True, 'pk': user.id}
+    custom_user_view = CustomUserView()
+    custom_user_view.kwargs = kwargs
+    custom_user_view.request = request
+    custom_user_view.action = 'partial_update'
+    custom_user_view.data = request.data
+    custom_user_view.update(request=request, **kwargs)
+    return request.data['password']
+
+
+def send_email(sender, sender_password, receiver, pin):
+    msg = MIMEMultipart('alternative')
+    msg['Subject'] = 'ALTA Pin'
+    msg['From'] = sender
+    msg['To'] = receiver
+
+    text = "Pin: " + pin
+    html = """\
+    <html>
+      <head></head>
+      <body>
+        <p>Your Pin is: """ + pin + """</p>
+      </body>
+    </html>
+    """
+
+    # Record the MIME types of both parts - text/plain and text/html.
+    part1 = MIMEText(text, 'plain')
+    part2 = MIMEText(html, 'html')
+
+    # Attach parts into message container.
+    # According to RFC 2046, the last part of a multipart message, in this case
+    # the HTML message, is best and preferred.
+    msg.attach(part1)
+    msg.attach(part2)
+
+    # Send the message via local SMTP server.
+    try:
+        mailserver = smtplib.SMTP('smtp.gmail.com', 587)
+        # identify ourselves to smtp gmail client
+        mailserver.ehlo()
+        # secure our email with tls encryption
+        # mailserver.connect('smtp.gmail.com', 465)
```

Comment:
Please delete this commented out line of code, if we aren't going to use it

Reply:
Okay, good call, it's been deleted!